.github/scripts/release.py

```diff
@@ -86,60 +110,148 @@ def printReleases(self):
         for r in releases:
             print(r)
 
-    def pushAutoCommits(self):
-        rc = push_git_tree_changes(self.repo_path, tag=self.tag, force_tag=True)
-        assert rc == 0, 'Failed to upload git tree changes'
+    def pushLocalCommits(self, force=False):
+        info('Pushing local commits...')
+        push_infos = self.local_repo.remote('origin').push(force=force)
+
+        # Check for any errors
+        for push_info in push_infos:
+            assert 0 == push_info.flags & PushInfo.ERROR, 'Failed to push changes to ' + str(push_info)
+
+    def pushTag(self):
+        # Overwrite existing tags
+        info('Pushing tag "%s"' % self.tag)
+        tag_info = self.local_repo.create_tag(self.tag, message=self.tag_msg, force=True)
+        self.local_repo.git.push(tags=True, force=True)
+
+    def deleteTag(self):
+        # Remove from remote
+        if self.tag in self.local_repo.tags:
+            info('Deleting tag "%s"' % self.tag)
+            self.local_repo.remote('origin').push(':%s' % self.tag)
+        else:
+            info('A tag does not exists for "%s". No need to delete.' % self.tag)
+
+    def updateSubmodulePointer(self, rel_path, ref):
+        submodule = Repo(rel_path)
+        submodule.remote('origin').fetch()
```

Comment:
We want to use the latest submodule tags, this could lead to a commit on a development branch I think